http-client/src/main/java/io/airlift/http/client/jetty/JettyClientDiagnostics.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.http.client.jetty;
+
+import com.google.common.util.concurrent.RateLimiter;
+import io.airlift.log.Logger;
+import org.eclipse.jetty.client.HttpClient;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import static java.util.Objects.requireNonNull;
+
+@ThreadSafe
+public class JettyClientDiagnostics
+{
+    private static final Logger log = Logger.get(JettyClientDiagnostics.class);
+
+    private final HttpClient httpClient;
```

Comment:
This doesn't need to be a field. You can just pass it into the logDiagnosticsInfo method.

If you do that, the object can be created where it's declared in JettyHttpClient and, in theory, could be shared for all clients.